chromium_src/components/search_engines/template_url_prepopulate_data.cc

```diff
@@ -53,61 +87,204 @@ const std::map<BravePrepopulatedEngineID, const PrepopulatedEngine*>
 std::vector<BravePrepopulatedEngineID> brave_engines_default = {
     PREPOPULATED_ENGINE_ID_GOOGLE,
     PREPOPULATED_ENGINE_ID_DUCKDUCKGO,
-#if defined(OS_ANDROID)
-    PREPOPULATED_ENGINE_ID_DUCKDUCKGO_LITE,
-#endif
     PREPOPULATED_ENGINE_ID_QWANT,
     PREPOPULATED_ENGINE_ID_BING,
     PREPOPULATED_ENGINE_ID_STARTPAGE,
 };
 
-// Germany - DDG appears on top.
+// Variations of the order / default options by country.
+std::vector<BravePrepopulatedEngineID> brave_engines_with_yahoo = {
+    PREPOPULATED_ENGINE_ID_YAHOO,
+    PREPOPULATED_ENGINE_ID_GOOGLE,
+    PREPOPULATED_ENGINE_ID_DUCKDUCKGO,
+    PREPOPULATED_ENGINE_ID_QWANT,
+    PREPOPULATED_ENGINE_ID_BING,
+    PREPOPULATED_ENGINE_ID_STARTPAGE,
+};
+
 std::vector<BravePrepopulatedEngineID> brave_engines_DE = {
+    PREPOPULATED_ENGINE_ID_YAHOO,
     PREPOPULATED_ENGINE_ID_DUCKDUCKGO_DE,
-#if defined(OS_ANDROID)
-    PREPOPULATED_ENGINE_ID_DUCKDUCKGO_LITE,
-#endif
     PREPOPULATED_ENGINE_ID_QWANT,
     PREPOPULATED_ENGINE_ID_GOOGLE,
     PREPOPULATED_ENGINE_ID_BING,
     PREPOPULATED_ENGINE_ID_STARTPAGE,
 };
 
-// France - Qwant appears on top.
 std::vector<BravePrepopulatedEngineID> brave_engines_FR = {
+    PREPOPULATED_ENGINE_ID_YAHOO,
     PREPOPULATED_ENGINE_ID_QWANT,
     PREPOPULATED_ENGINE_ID_GOOGLE,
     PREPOPULATED_ENGINE_ID_DUCKDUCKGO,
-#if defined(OS_ANDROID)
-    PREPOPULATED_ENGINE_ID_DUCKDUCKGO_LITE,
-#endif
     PREPOPULATED_ENGINE_ID_BING,
     PREPOPULATED_ENGINE_ID_STARTPAGE,
 };
 
-// Australia / NZ / IE - DDG appears on top.
 std::vector<BravePrepopulatedEngineID> brave_engines_AU_NZ_IE = {
+    PREPOPULATED_ENGINE_ID_YAHOO,
     PREPOPULATED_ENGINE_ID_DUCKDUCKGO_AU_NZ_IE,
-#if defined(OS_ANDROID)
-    PREPOPULATED_ENGINE_ID_DUCKDUCKGO_LITE,
-#endif
     PREPOPULATED_ENGINE_ID_GOOGLE,
     PREPOPULATED_ENGINE_ID_QWANT,
     PREPOPULATED_ENGINE_ID_BING,
     PREPOPULATED_ENGINE_ID_STARTPAGE};
 
-// A map to keep track of default engines for countries that don't use the
-// regular default engine.
+// A map to keep track of a full list of default engines for countries
+// that don't use the default list.
 const std::map<int, std::vector<BravePrepopulatedEngineID>>
     default_engines_by_country_id_map = {
```

Comment:
can we hold just pointers? Keeping ~30 copies of the same vector doesn't look super efficient